commit 43b1084d52892fffea7f2e6cd3731ad2a46b52bd
Author: Ray Strode <rstrode@redhat.com>
Date:   Thu May 7 16:34:57 2009 -0400

    Set is_authenticated to TRUE when authenticated
    
    Before the flag was never getting set causing btmp records to get
    written when the session and slave are killed around the same time
    (by switching runlevels)

diff --git a/daemon/gdm-session-direct.c b/daemon/gdm-session-direct.c
index 34a126f..7259bf6 100644
--- a/daemon/gdm-session-direct.c
+++ b/daemon/gdm-session-direct.c
@@ -334,6 +334,7 @@ gdm_session_direct_handle_authenticated (GdmSessionDirect *session,
         dbus_connection_send (connection, reply, NULL);
         dbus_message_unref (reply);
 
+        session->priv->is_authenticated = TRUE;
         _gdm_session_authenticated (GDM_SESSION (session));
 
         return DBUS_HANDLER_RESULT_HANDLED;
@@ -361,6 +362,7 @@ gdm_session_direct_handle_authentication_failed (GdmSessionDirect *session,
 
         g_debug ("GdmSessionDirect: Emitting 'authentication-failed' signal");
 
+        session->priv->is_authenticated = FALSE;
         _gdm_session_authentication_failed (GDM_SESSION (session), NULL);
 
         return DBUS_HANDLER_RESULT_HANDLED;
